src/login.rs

```diff
@@ -73,3 +73,28 @@ pub fn get_cgroup(pid: Option<pid_t>) -> Result<String> {
     let cg = unsafe { MString::from_raw(c_cgroup) };
     Ok(cg.unwrap().to_string())
 }
+
+/// Determines the session identifier of a process.
+///
+/// Specific processes can be optionally targeted via their PID. When no PID is
+/// specified, operation is executed for the calling process.
+/// This method can be used to retrieve a session identifier.
+pub fn get_session(pid: Option<pid_t>) -> Result<String> {
+    let mut c_session: *mut c_char = ptr::null_mut();
+    let p: pid_t = pid.unwrap_or(0);
```

Comment:
While the existing code does do this (so I think keeping it for now is OK) I'm not too sure about using `Option<pid_t>` here (or elsewhere in the existing code). `pid_t` can already be zero, so all we're really doing here is increasing code & argument size by wrapping it in an `Option<>`.

Something to think about for a major version bump I guess.